code/game/objects/items/tools/experimental_tools.dm

```diff
@@ -121,3 +121,181 @@
 	dat += "</table>"
 	dat += "<br><hr>"
 	return dat
+
+/obj/item/tool/portadialysis
+	name = "portable dialysis machine"
+	desc = "A man-portable dialysis machine, with a small internal battery that can be recharged. Filters out all foreign compounds from the bloodstream of whoever it's attached to, but also typically ends up removing some blood as well."
+	icon = 'icons/obj/items/experimental_tools.dmi'
+	icon_state = "portadialysis"
+	item_state = "syringe_0"
+	flags_equip_slot = SLOT_WAIST
+	w_class = SIZE_MEDIUM
+	var/attaching = FALSE
+	var/filtering = FALSE
+	var/mob/living/carbon/human/attached = null
+	var/reagent_removed_per_second = AMOUNT_PER_TIME(3, 2 SECONDS)
+	var/obj/item/cell/pdcell = null
+	var/filter_cost = AMOUNT_PER_TIME(20, 2 SECONDS)
+	var/blood_cost = AMOUNT_PER_TIME(12, 2 SECONDS)
+
+/obj/item/tool/portadialysis/Initialize(mapload, ...)
+	. = ..()
+
+	pdcell = new/obj/item/cell(src) //has 1000 charge
+	update_icon()
+
+/obj/item/tool/portadialysis/update_icon()
+	if(attaching)
+		flick("portadialysis_starting", src)
+
+	else if(filtering)
+		icon_state = "portadialysis_running"
+
+	else
+		flick("portadialysis", src) //we do this to override the previous flick()
+		icon_state = "portadialysis"
+
+	if(pdcell && pdcell.charge)
+		overlays.Cut()
+		switch(round(pdcell.charge * 100 / pdcell.maxcharge))
+			if(85 to INFINITY)
+				overlays += "dialysis_battery_100"
+			if(60 to 84)
+				overlays += "dialysis_battery_85"
+			if(45 to 59)
+				overlays += "dialysis_battery_60"
+			if(30 to 44)
+				overlays += "dialysis_battery_45"
+			if(15 to 29)
+				overlays += "dialysis_battery_30"
+			if(1 to 14)
+				overlays += "dialysis_battery_15"
+
+/obj/item/tool/portadialysis/examine(mob/user)
+	..()
+	var/currentpercent = 0
+	currentpercent = round(pdcell.charge * 100 / pdcell.maxcharge)
+	to_chat(user, SPAN_INFO("It has [currentpercent]% charge left in its internal battery."))
+
+/obj/item/tool/portadialysis/attack(mob/living/carbon/human/target, mob/living/carbon/human/user)
+	if(!isHumanStrict(target))
+		return ..()
+
+	if(!skillcheck(user, SKILL_MEDICAL, SKILL_MEDICAL_MEDIC))
+		to_chat(user, SPAN_WARNING("You don't seem to know how to use \the [src]..."))
+		return
+
+	if(!pdcell || pdcell.charge == 0)
+		to_chat(user, SPAN_NOTICE("\The [src] flashes its 'battery low' light, and refuses to attach."))
+		return
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
```

Comment:
don't need H here, user is already typecasted up to human

Reply:
also H isn't a detailed enough variable name anyways

Reply:
changed to user